sct_ssh.py

```diff
@@ -433,3 +443,99 @@ def gcp_allow_public(user, test_id):
+        sys.exit(1)
+    if packages_location.startswith('s3') or packages_location.startswith('gs'):
+        packages_location = download_dir_from_cloud(packages_location)
+    if not packages_location.endswith('/'):
+        packages_location += '/'
+
+    instances = select_instance_group(region=region, backends=[backend], test_id=test_id)
+    for instance in instances:
+        tags = get_tags(instance)
+        instance.update({
+            'NodeType': tags['NodeType'],
+            'Name': tags['Name'],
+            'User': tags['RunByUser']
+        })
+
+    if any([i for i in instances if 'db' not in i['NodeType']]):
+        LOGGER.error("Only DB nodes should be selected for updating scylla packages")
+        sys.exit(1)
+
+    nodes = []
+    for instance in instances:
+        remoter = RemoteCmdRunnerBase.create_remoter(
+            hostname=instance['PublicIpAddress'],
+            user=guess_username(instance),
+            key_file=f"~/.ssh/{instance['KeyName']}")
+        nodes.append(remoter)
+
+    def run_func_parallel(func, node_list):
+        with ThreadPoolExecutor() as executor:
+            futures = [executor.submit(func, node) for node in node_list]
+            results = [future.result() for future in as_completed(futures)]
+
+        return results
+
+    def is_scylla_active(node: RemoteCmdRunnerBase):
+        return node.run('systemctl is-active scylla-server.service', ignore_status=True).stdout.strip() == 'active'
+
+    def stop_scylla(node: RemoteCmdRunnerBase):
+        node.sudo('systemctl stop scylla-server.service')
+        wait.wait_for(func=lambda: not is_scylla_active(node), step=10, timeout=180)
+
+    def start_scylla(node: RemoteCmdRunnerBase):
+        node.sudo('systemctl start scylla-server.service')
+        wait.wait_for(func=lambda: is_scylla_active(node), step=10, timeout=180)
+
+    def update_packages(node: RemoteCmdRunnerBase):
+        node.run('mkdir -p /tmp/scylla')
+        node.send_files(packages_location, '/tmp/scylla')
+        node.run('tar -xvf /tmp/scylla/*.tar.gz -C /tmp/scylla/', ignore_status=True)
+
+        def check_package_extension(node, extension):
+            fit_extension = node.run(f'ls /tmp/scylla/*.{extension}', ignore_status=True)
+            if not fit_extension.ok:
+                LOGGER.error("There is no suitable packages for this distro on the node" % node.hostname)
+                sys.exit(1)
+
+        distro = Distro.from_os_release(node.run("cat /etc/os-release", ignore_status=True, retry=5).stdout)
+        if distro.is_rhel_like:
+            check_package_extension(node, 'rpm')
+            node.run('yum list installed | grep scylla')
+            node.sudo('rpm -URvh --replacefiles /tmp/scylla/*.rpm')
+            node.run('yum list installed | grep scylla')
+        elif distro.is_ubuntu:
+            check_package_extension(node, 'deb')
+            node.sudo(shell_script_cmd("yes Y | dpkg --force-depends -i /tmp/scylla/scylla*"))
```

Comment:
in case of failure, is it properly reported to console for debugging?

Reply:
Yes
It is reported similarly to how it is done in `_update_db_packages` of the BaseScyllaCLuster, e.g.:
```
10:26:26  Command: 'sudo bash -cxe "yes Y | dpkg --force-depends -i /tmp/scylla/scylla*"'
10:26:26  
10:26:26  Exit code: 1
10:26:26  
10:26:26  Stdout:
10:26:26  
10:26:26  Preparing to unpack .../scylla-enterprise-python3_2024.2.0~dev-0.20240127.5f89014c0f87-1_amd64.deb ...
...
10:26:26  Stderr:
10:26:26  
10:26:26    Version of scylla-enterprise-kernel-conf on system is 2023.1.9-0.20240609.4a93c32572b9-1.
10:26:26   scylla-enterprise depends on scylla-enterprise-cqlsh (= 2024.2.0~dev-0.20240127.5f89014c0f87-1); however:
10:26:26    Package scylla-enterprise-cqlsh is not installed.
...
```